src/wp-includes/rest-api.php

```diff
@@ -1697,8 +1930,12 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 		$best_type = rest_handle_multi_type_schema( $value, $args, $param );
 
 		if ( ! $best_type ) {
-			/* translators: 1: Parameter, 2: List of types. */
-			return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s is not of type %2$s.' ), $param, implode( ',', $args['type'] ) ) );
+			return new WP_Error(
+				'rest_invalid_type',
+				/* translators: 1: Parameter, 2: List of types. */
+				sprintf( __( '%1$s is not of type %2$s.' ), $param, implode( ',', $args['type'] ) ),
+				array( 'param' => $param )
+			);
```

Comment:
Is there a back compat concern around changing this error code?

Reply:
Will definitely call it out in the dev note, but I think we're ok. The `rest_invalid_param` error is the most generic error codes that we use. So someone wouldn't be able to use the error code alone to tell that it is a specific error. They'd have to do analysis on the error message itself.

We also don't currently expose the error codes to API consumers. It would only impact people using the REST validation function directly.